### ISSUE TITLE
Add LocalizableRandomAccess that provides access to the position and value supplier

### ISSUE THREAD

**Issue by cmhulbert** [imglib/imglib2#324]
Reference: [Localizables](https://github.com/imglib/imglib2/blob/b51f41a505aa75ed8ac861f530d4727b695aea6a/src/main/java/net/imglib2/util/Localizables.java)

Should provide the position without a copy, and the value supplier should not be evalutated unless explicitly called after the `get` on the LocalizableRandomAcess

**Comment by tpietzsch** [comment]
this is a bit too cryptic for me ... what is a usage example you envision for this LocalizableRandomAccess?

**Comment by cmhulbert** [comment]
Sorry I was a bit hasty when I wrote this. I wrote this with @axtimwalde, so he may be able to provide additional context as well.

Wanting something like this came up when trying to use `LoopBuilder` over a set of images, and have access to both the values, and the positions they are at. Using the current LocalizableRandomAccess should make this possible, but then you are moving two cursors for each pixel, one for the values, and one for the positions. This proposal was to provide a variant of LocalizableRandomAcess which would have a `get` that returns a `Pair<Localizable, Supplier<T>>`. This allows us to use the position without having to move a duplicate cursor, as well as have access to the value at that location, but _only_ if needed, hence the supplier.
